src/overlays/actors/ovl_En_Kusa/z_en_kusa.c

```diff
@@ -7,72 +16,485 @@
+    ICHAIN_VEC3F_DIV1000(scale, 400, ICHAIN_CONTINUE),         ICHAIN_F32_DIV1000(gravity, -3200, ICHAIN_CONTINUE),
+    ICHAIN_F32_DIV1000(minVelocityY, -17000, ICHAIN_CONTINUE), ICHAIN_F32(uncullZoneForward, 1200, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneScale, 100, ICHAIN_CONTINUE),         ICHAIN_F32(uncullZoneDownward, 120, ICHAIN_STOP),
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kusa/func_80A9AFA0.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kusa/func_80A9AFAC.s")
+s32 D_80A9C294[] = { 0x0500B9D0, 0x06000140, 0x06000140 };
+
+extern Gfx D_060002E0[];
+extern Gfx D_040355E0[]; // bush fragments 1
+extern Gfx D_040356A0[]; // bush fragments 2
+
+void func_80A9AFA0(EnKusa* this, EnKusaActionFunc actionFunc) {
+    this->unk_19C = 0;
+    this->actionFunc = actionFunc;
+}
+
+// Snap to floor
+s32 func_80A9AFAC(EnKusa* this, GlobalContext* globalCtx, f32 arg2) {
+    CollisionPoly* sp34;
+    f32 sp28;
+    Vec3f sp30;
+    UNK_TYPE sp24;
+    f32 temp_f0;
+
+    sp30.x = this->actor.posRot.pos.x;
+    sp30.y = this->actor.posRot.pos.y + 30.0f;
+    sp30.z = this->actor.posRot.pos.z;
+    temp_f0 = func_8003C9A4(&globalCtx->colCtx, &sp28, &sp24, &this->actor, &sp30);
+    if (-32000.0f < temp_f0) {
+        this->actor.posRot.pos.y = temp_f0 + arg2;
+        Math_Vec3f_Copy(&this->actor.initPosRot, &this->actor.posRot);
+        return true;
+    }
+    osSyncPrintf(VT_COL(YELLOW, BLACK));
+    // Translation: Failure attaching to ground
+    osSyncPrintf("地面に付着失敗(%s %d)\n", "../z_en_kusa.c", 0x143);
+    osSyncPrintf(VT_RST);
+    return false;
+}
+
+// Spawn Collectible
+void func_80A9B07C(EnKusa* this, GlobalContext* globalCtx) {
+    s16 dropParams;
+    switch (this->actor.params & 3) {
+        case 0:
+        case 2:
+            dropParams = (this->actor.params >> 8) & 0xF;
+            if (dropParams >= 0xD) {
+                dropParams = 0;
+            }
+            Item_DropCollectibleRandom(globalCtx, NULL, &this->actor.posRot.pos, dropParams << 4);
+            break;
+
+        case 1:
+            if (Math_Rand_ZeroOne() < 0.5f) {
+                Item_DropCollectible(globalCtx, &this->actor.posRot.pos, 0x10);
+
+            } else {
+                Item_DropCollectible(globalCtx, &this->actor.posRot.pos, 3);
+                break;
+            }
+    }
+}
+
+void func_80A9B140(EnKusa* this) {
+    this->actor.velocity.y += this->actor.gravity;
+
+    if (this->actor.velocity.y < this->actor.minVelocityY) {
+        this->actor.velocity.y = this->actor.minVelocityY;
+    }
+}
+
+void func_80A9B174(Vec3f* this, f32 arg1) {
+
+    arg1 += ((Math_Rand_ZeroOne() * 0.2f) - 0.1f) * arg1;
+
+    this->x -= this->x * arg1;
+    this->y -= this->y * arg1;
+    this->z -= this->z * arg1;
+}
+
+// Scale factor set/init?
+void func_80A9B1FC(EnKusa* this) {
+    this->actor.scale.y = 0.16000001f;
+    this->actor.scale.x = 0.120000005f;
+    this->actor.scale.z = 0.120000005f;
```

Comment:
Can these be `0.16` and `0.12`?

Reply:
From what I tried, it didn't match after changing them